### PR TITLE
Merge 2.4 into develop

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Merge 2.4 into develop, picking up these PRs:
https://github.com/juju/juju/pull/9287 container networking, manually provisioned machines
https://github.com/juju/juju/pull/9296 more robust cmr controller connectivity
https://github.com/juju/juju/pull/9251 Fix the exporting of the series of the application when exporting a bundle
https://github.com/juju/juju/pull/9293 bundle deploy panic
https://github.com/juju/juju/pull/9285 tomb.Go shouldn't be called multiple times from a non-tomb managed goroutine
https://github.com/juju/juju/pull/9263 Add back the state of each unit in a relation to goal state
https://github.com/juju/juju/pull/9235 handle maas partitions in a storage constraint
https://github.com/juju/juju/pull/9191 Add a default retry inside 'juju status'
https://github.com/juju/juju/pull/9177 latest ec2 instance types
https://github.com/juju/juju/pull/9180 fix space check before upgrade
https://github.com/juju/juju/pull/9159 mongo collection size

## QA steps

bootstrap smoke test

### CODE REVIEW

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

File: Gopkg.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@
   name = "github.com/hashicorp/raft-boltdb"
 
 [[constraint]]
-  revision = "bc5b7e5db16683db8ddffa7eef67b53706a4618a"
+  revision = "735c14e2335ad2d8112482794d69629c0517476c"
```
Comment (anastasiamac) [inline Gopkg.toml]: I do not think we want to change these here!Isn't it all revisions from 2.4?

Comment (wallyworld) [reply Gopkg.toml]: 2.4 uses dependencies.tsv instead of Gopkg.toml
These shas have been ported across from the changes to dependencies.tsv

Comment (anastasiamac) [reply Gopkg.toml]: Niice pick up :) Thank you!

**Pull request merged**